Add examples on how to embed a Flutter screen into an existing SwiftUI app (#7828)
flutter/website: src/development/add-to-app/ios/project-setup.md

description: Learn how to integrate a Flutter module into your existing iOS project.
---

Flutter UI components can be incrementally added into your existing iOS
application as embedded frameworks. There are a few ways to embed Flutter 
in your existing application.

1. **Use the CocoaPods dependency manager and installed Flutter SDK.**
  In this case, the `flutter_module` is compiled from
  the source each time the app is built. (Recommended.)

1. **Create frameworks for the Flutter engine, your compiled Dart code,
  and all Flutter plugins.** Here, you manually embed the frameworks,
  and update your existing application's build settings in Xcode.
  This can be useful for teams that don't want to require every developer
  to have the Flutter SDK and Cocoapods installed locally.

1. **Create frameworks for your compiled Dart code,

>   and all Flutter plugins. Create a Pod library for the Flutter engine.** 

Reviewer: I don't think "Pod library" is quite right, maybe:
```suggestion
  and all Flutter plugins. Use CocoaPods for the Flutter engine.** 
```

>   With this option,  you embed the frameworks for your application

Reviewer: ```suggestion
  With this option, embed the frameworks for your application
```

  and the plugins in Xcode, but distribute the

>   Flutter engine as a Cocoapods podspec.

Reviewer: ```suggestion
  Flutter engine as a CocoaPods podspec.
```

## Embed the Flutter module in your existing application

After you have developed your Flutter module,
you can embed it using the methods described at the top of the page.

{{site.alert.note}}
  You can run in Debug mode on a simulator or a real device,

>   and Release on a real device. Learn more at 

Reviewer 2: ```suggestion
  and Release on a real device. Learn more about
```

```

If your existing application (`MyApp`) doesn't
already have a Podfile, run `pod init` in the  
`MyApp` directory to create one. 
You can find more details on using 

> Cocoapods in the [CocoaPods getting started guide][].

Reviewer 2: ```suggestion
CocoaPods in the [CocoaPods getting started guide][].
```

```

{{site.alert.note}}

>   The `flutter_post_install` method (added in Flutter 3.1.0),

Reviewer: Thanks for changing this.

Author: No prob!

You must run `flutter build ios-framework`
every time you make code changes in your Flutter module.

The following example assumes that you want to generate the

> frameworks to `some/path/Flutter/`.

Reviewer: I kind of liked the old `MyApp` directory so you know it should be relative or inside your host app in a known location, not just floating somewhere in another directory.

Author: The reason why I moved it out was because I thought it wouldn't be directly inside your app codebase because I figured if other developers didn't have dart / flutter on their computer then it wouldn't make sense for them to bring the code in when they're working on the app. But I can definitely add it back!

Reviewer: This is the path to the built Flutter/App/plugin frameworks, not the Flutter module containing the dart code.  Without these frameworks the app won't build, so they need to make it available to other developers that don't have Flutter installed somehow.

Author: Ahh right, okay makes sense -- I'll change back!

[VS Code]: {{site.url}}/development/tools/vs-code
[XCFrameworks]: {{site.apple-dev}}/documentation/xcode_release_notes/xcode_11_release_notes
[Xcode installed]: {{site.url}}/get-started/install/macos#install-xcode

> [News Feed app]: https://github.com/flutter/put-flutter-to-work/tree/022208184ec2623af2d113d13d90e8e1ce722365

Reviewer 3: do we want po provide 2 examples? (one with app delegate and the other without)

Author: I imagine that someone could adapt the example based on the inline code we provide, but I'm sure it wouldn't hurt to have more examples. I don't think it makes sense to prioritize creating a full example right now, but maybe we can revisit after some of the other things (extensions, companion app, etc) have been added? WDYT?

Reviewer 3: sounds good
